### PR TITLE
[ios] Added annotation selection tests with rotation & scaling of the map

### PR DESCRIPTION
**Pull request by julianrex**
WIP - needs some cleanup, but ok for an initial review.

Adds integration tests to check annotation selection after rotating or zooming the map (not via a gesture).

### CODE REVIEW

**Comment by julianrex** [comment]
@astojilj in the light of https://github.com/mapbox/mapbox-gl-native/pull/15130 - how would you recommended I update this PR?

File: platform/ios/src/MGLMapView_Private.h [hunk 1]
```diff
@@ -52,8 +52,6 @@ FOUNDATION_EXTERN MGL_EXPORT MGLExceptionName const _Nonnull MGLUnderlyingMapUna
 /// Synchronously render a frame of the map.
 - (void)renderSync;
 
-- (nonnull mbgl::Map *)mbglMap;
```
Comment (astojilj) [inline platform/ios/src/MGLMapView_Private.h]: This seems orthogonal to other changes. Why this?

File: platform/ios/Integration Tests/Annotation Tests/MGLAnnotationViewIntegrationTests.mm [hunk 2]
```diff
@@ -500,6 +511,248 @@ - (void)testUserLocationWithOffsetAnchorPoint {
...
+    }
+    coordinates[20] = CLLocationCoordinate2DMake(NAN, NAN);
+
+    NSArray *annotations = [self internalAddAnnotationsAtCoordinates:coordinates];
+
+    XCTestExpectation *showCompleted = [self expectationWithDescription:@"showCompleted"];
+
+    CGSize size = self.mapView.frame.size;
+    
+    CGFloat maximumPadding = 100;
```
Comment (astojilj) [inline platform/ios/Integration Tests/Annotation Tests/MGLAnnotationViewIntegrationTests.mm]: Let's make insets asymmetric (e.g. use just top and left) and larger than 100:
e.g. UIEdgeInsetsMake(size.height / 2, size.width / 3, 0, 0);

Asymmetric because we want the content center to be away from screen center.

**Comment by astojilj** [comment]
> @astojilj in the light of #15130 - how would you recommended I update this PR?

Sorry for late response. Covered in https://github.com/mapbox/mapbox-gl-native/pull/15123#discussion_r311241680

File: platform/ios/src/MGLMapView_Private.h (hunk 1 above)
Comment (julianrex) [reply platform/ios/src/MGLMapView_Private.h]: Moved this to another PR.

File: platform/ios/Integration Tests/Annotation Tests/MGLAnnotationViewIntegrationTests.mm (hunk 2 above)
Comment (julianrex) [reply platform/ios/Integration Tests/Annotation Tests/MGLAnnotationViewIntegrationTests.mm]: done.
